src/main/java/net/mcreator/ui/component/JStringListField.java

```diff
@@ -0,0 +1,202 @@
+		super(new BorderLayout());
+		setBackground((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT"));
+
+		label.setOpaque(false);
+		ComponentUtils.deriveFont(label, 16);
+		ToolTipManager.sharedInstance().registerComponent(label);
+
+		JScrollPane scrollPane = new JScrollPane(label);
+		scrollPane.setOpaque(false);
+		scrollPane.getViewport().setOpaque(false);
+		scrollPane.setPreferredSize(new Dimension(200, 30));
+
+		back.setMargin(new Insets(0, 0, 0, 0));
+		back.setBorder(BorderFactory.createEmptyBorder(0, 5, 0, 5));
+		back.setFocusPainted(false);
+		back.setContentAreaFilled(false);
+		back.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+		back.addActionListener(e -> {
+			if (!joinEntries) {
+				if (index <= 0)
+					index = textList.size();
+				index--;
+				refreshVisibleText();
+			}
+		});
+
+		forward.setMargin(new Insets(0, 0, 0, 0));
+		forward.setBorder(BorderFactory.createEmptyBorder(0, 5, 0, 5));
+		forward.setFocusPainted(false);
+		forward.setContentAreaFilled(false);
+		forward.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+		forward.addActionListener(e -> {
+			if (!joinEntries) {
+				index++;
+				if (index >= textList.size())
+					index = 0;
+				refreshVisibleText();
+			}
+		});
+
+		edit.setOpaque(false);
+		edit.setMargin(new Insets(0, 0, 0, 0));
+		edit.setBorder(BorderFactory.createEmptyBorder());
+		edit.setContentAreaFilled(false);
+		edit.setToolTipText(L10N.t("components.string_list.edit"));
+		edit.addActionListener(e -> {
+			List<String> newList = ListEditorDialog.open(parent, textList, validator, uniqueEntries);
+			if (newList != null)
+				setTextList(newList);
+		});
+
+		JButton copy = new JButton(UIRES.get("16px.copyclipboard")) {
+			@Override public String getName() {
+				return "TechnicalButton";
+			}
+		};
+		copy.setOpaque(false);
+		copy.setMargin(new Insets(0, 0, 0, 0));
+		copy.setBorder(BorderFactory.createEmptyBorder());
+		copy.setContentAreaFilled(false);
+		copy.setToolTipText(L10N.t("components.string_list.copy"));
+		copy.addActionListener(e -> Toolkit.getDefaultToolkit().getSystemClipboard()
+				.setContents(new StringSelection(String.join(System.lineSeparator(), textList)), null));
+
+		JPanel controls = new JPanel();
+		controls.setBackground((Color) UIManager.get("MCreatorLAF.DARK_ACCENT"));
+		controls.add(edit);
+		controls.add(copy);
+
+		add("West", back);
+		add("Center", scrollPane);
+		add("East", PanelUtils.westAndEastElement(forward, PanelUtils.centerInPanel(controls)));
+	}
+
+	@Override public void setEnabled(boolean b) {
+		super.setEnabled(b);
+		edit.setEnabled(b);
+	}
+
+	/**
+	 * @param joinEntries Whether string entries should be joined using commas and shown together.
+	 * @return This field instance.
+	 */
+	public JStringListField setJoinEntries(boolean joinEntries) {
+		this.joinEntries = joinEntries;
+		back.setVisible(!joinEntries);
+		forward.setVisible(!joinEntries);

```

Comment:
Call to this method should also update UI (refreshVisibleText?) so change is dynamic if needed